skyvern/webeye/persistent_sessions_manager.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import annotations
+
+import asyncio
+import socket
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple
+
+import structlog
+from playwright.async_api import async_playwright
+
+from skyvern.forge.sdk.db.client import AgentDB
+from skyvern.forge.sdk.schemas.persistent_browser_sessions import PersistentBrowserSession
+from skyvern.forge.sdk.schemas.tasks import ProxyLocation
+from skyvern.webeye.browser_factory import BrowserContextFactory, BrowserState
+
+LOG = structlog.get_logger()
+
+
+@dataclass
+class BrowserSession:
+    browser_state: BrowserState
+    cdp_port: int
+    cdp_host: str = "localhost"
+
+
+class PersistentSessionsManager:
+    instance = None
+    _browser_sessions: Dict[str, BrowserSession] = dict()
+
+    def __init__(self, database: AgentDB):
+        self.database = database
+
+    def __new__(cls, database: AgentDB) -> PersistentSessionsManager:
+        if cls.instance is None:
+            cls.instance = super().__new__(cls)
+            cls.instance.database = database
+        return cls.instance
+
+    async def get_active_sessions(self, organization_id: str) -> List[PersistentBrowserSession]:
+        """Get all active sessions for an organization."""
+        return await self.database.get_active_persistent_browser_sessions(organization_id)
+
+    def get_browser_state(self, session_id: str) -> Optional[BrowserState]:
```

Comment:
Can do this after python 3.9:
```suggestion
    def get_browser_state(self, session_id: str) -> BrowserState | None:
```